### DISCUSSION TITLE
[Question]: Detect last sector of used memory

### DISCUSSION THREAD

**Discussion by hasenradball** [Drewsif/PiShrink#291] [General]
Hi,

I wanted to ask you how you detect the last sector or Byte which is used, to be able to remove the unused Bytes.

Point is I have the problem, that I use a 256 GB ssd for pi and don‘ t want to make an image of the whole ssd.
When only 18 GB of the disk is used.

**Comment by Drewsif** [comment]
I use `resize2fs` to shrink the ext filesystem (https://github.com/Drewsif/PiShrink/blob/v24.10.10/pishrink.sh#L350). Then `parted` is used to shrink the partition to its smallest size (https://github.com/Drewsif/PiShrink/blob/v24.10.10/pishrink.sh#L374). The links are to the relevant code with the commands and math on how to do it. Then you could just `dd` up until the last partition ends.
